R/regression.R

```diff
@@ -2315,3 +2317,34 @@ hccmAdjust <- function(fit.reg, robust.se, h)
     factor[h == 1] <- df.res/n
     V %*% t(X) %*% apply(X, 2, "*", (e^2)/factor) %*% V
 }
+
+reduceOutputSize <- function(fit)
+{
+    original <- fit$original
+    ## Can't delete qr; needed for predict, rstudent, lm.influence, outlierTest
+    ## original$qr$qr <- c()
+    ## remove environment attribute to reduce size
+    attr(fit$terms, ".Environment") <- c()
+    attr(original$terms, ".Environment") <- c()
+    attr(original$formula, ".Environment") <- c()
+    attr(attr(original$model, "terms"), ".Environment") <- c()
+    attr(fit$summary$formula, ".Environment") <- c()
+    attr(fit$summary$terms, ".Environment") <- c()
+    ## remove names from residuals and fitted values to reduce size
+    original$residuals <- unname(original$residuals)
+    original$fitted.values <- unname(original$fitted.values)
+    original$weights <- unname(original$weights)
+    original$prior.weights <- unname(original$prior.weights)
+    if (is.matrix(fit$summary$fitted.values))
+        rownames(fit$summary$fitted.values) <- NULL
+    if (is.vector(original$y))
+        original$y <- unname(original$y)
+    if (is.vector(original$linear.predictors))
+        original$linear.predictors <- unname(original$linear.predictors)
+    ## if (!inherits(original, "svyglm"))
+    ##     original$y <- NULL
+    ## original$survey.design <- NULL  # needed can't delete
+    original$data <- NULL
+    fit$original <- original
+    return(fit)
```

Comment:
Consider adding code here to check if it is a weighted regression and possibly copy the appropriate design list slot by reference. (See overall comment)